demos/common/sample_azure_common.h

```diff
@@ -0,0 +1,26 @@
+/* Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License. */
+
+/**
+ * @brief Contains helpers that are used in multiple samples
+ */
+
+#ifndef SAMPLE_AZURE_COMMON_IF_H
+#define SAMPLE_AZURE_COMMON_IF_H
+
+/* Demo Specific configs. */
+#include "demo_config.h"
+
+#if NDEBUG
+    #define panic_handler()                                                                           \
+    vLoggingPrintf( "[ERROR] [%s:%s:%d]\r\nRestarting device...\r\n", __FILE__, __func__, __LINE__ ); \
+    configRestartDevice()
+#else
+    #define panic_handler()                                                   \
+    vLoggingPrintf( "[ERROR] [%s:%s:%d]\r\n", __FILE__, __func__, __LINE__ ); \
+    vLoggingPrintf( "Looping to enable attaching a debugger\r\n" );           \
+    taskDISABLE_INTERRUPTS();                                                 \
```

Comment:
Why do we disable interrupts here? Our code shouldn't really ever need to create a critical section (instead we can use OS-level constructs such as mutexes, semaphores, etc).

Reply:
I was following FreeRTOS's existing configAssert implementation, but I'm open to other solutions that would enable attaching a debugger the best!

Reply:
removed the line to disable interrupts - let me know if there was something else you had in mind!